File: tests/benchmark/corpus/go/path_pruning/safe_early_return.go
package main

import (
	"net/http"
	"os/exec"
)

var allowed = map[string]bool{"status": true, "version": true, "uptime": true}

func handler(w http.ResponseWriter, r *http.Request) {
	cmd := r.URL.Query().Get("cmd")
	if !allowed[cmd] {
		http.Error(w, "forbidden", 403)
		return
	}
	exec.Command(cmd).Run()
}

File: tests/benchmark/corpus/javascript/path_pruning/safe_early_return.js
const express = require('express');
const { exec } = require('child_process');
const app = express();

const ALLOWED = ['status', 'version', 'uptime'];

app.get('/run', (req, res) => {
    const cmd = req.query.cmd;
    if (!ALLOWED.includes(cmd)) {
        return res.status(403).send('forbidden');
    }
    exec(cmd);
});
